test/jdk/jdk/classfile/StrictStackMapsTest.java

```diff
@@ -0,0 +1,260 @@
+                        .invokespecial(CD_Object, INIT_NAME, MTD_void)
+                        .return_()));
+        var clazz = ByteCodeLoader.load(className, classBytes); // sanity check to pass verification
+        var classModel = ClassFile.of().parse(classBytes);
+        var ctorModel = classModel.methods().getFirst();
+        var stackMaps = ctorModel.code().orElseThrow().findAttribute(Attributes.stackMapTable()).orElseThrow();
+        assertEquals(2 * 2, stackMaps.entries().size()); // if -> else, then -> end of if + asserts
+        var elseAssertFrame = stackMaps.entries().get(0); // else jump from if
+        assertEquals(246, elseAssertFrame.frameType());
+        assertEquals(List.of(ConstantPoolBuilder.of().nameAndTypeEntry("fsf", CD_int)), elseAssertFrame.unsetFields());
+        var mergedAssertFrame = stackMaps.entries().get(2); // then jump to join else
+        assertEquals(246, mergedAssertFrame.frameType());
+        assertEquals(List.of(), mergedAssertFrame.unsetFields());
+    }
+
+    @Test
+    void skipUnnecessaryUnsetFramesTest() throws Throwable {
+        var className = "Test";
+        var classDesc = ClassDesc.of(className);
+        var classBytes = ClassFile.of().build(classDesc, clb -> clb
+                .withField("fPlain", CD_char, ACC_PRIVATE)
+                .withField("fs", CD_int, ACC_STRICT)
+                .withField("fsf", CD_int, ACC_STRICT | ACC_FINAL)
+                .withMethodBody(INIT_NAME, MTD_void, 0, cob -> cob
+                        .aload(0)
+                        .iconst_5()
+                        .putfield(classDesc, "fs", CD_int)
+                        .aload(0)
+                        .iconst_0()
+                        .ifThenElse(thb -> thb
+                                .iconst_3()
+                                .putfield(classDesc, "fPlain", CD_int), elb -> elb
+                                .iconst_2()
+                                .putfield(classDesc, "fPlain", CD_int))
+                        .aload(0)
+                        .iconst_5()
+                        .putfield(classDesc, "fsf", CD_int)
+                        .aload(0)
+                        .invokespecial(CD_Object, INIT_NAME, MTD_void)
+                        .return_()));
+        var clazz = ByteCodeLoader.load(className, classBytes); // sanity check to pass verification
+        var classModel = ClassFile.of().parse(classBytes);
+        var ctorModel = classModel.methods().getFirst();
+        var stackMaps = ctorModel.code().orElseThrow().findAttribute(Attributes.stackMapTable()).orElseThrow();
+        assertEquals(2 + 1, stackMaps.entries().size()); // if -> else, then -> end of if, just one assert
+        var assertFrame = stackMaps.entries().get(0); // else jump from if
+        assertEquals(246, assertFrame.frameType());
+        assertEquals(List.of(ConstantPoolBuilder.of().nameAndTypeEntry("fsf", CD_int)), assertFrame.unsetFields());
+    }
+
+    @Test
```

Comment:
nit: it would be great if a comment with the corresponding java code being generated was included in these tests, this will make easier future changes, fixing bugs etc. But I let that to your consideration